omod/src/main/webapp/resources/scripts/custom/deletePatient.js

```diff
@@ -0,0 +1,61 @@
+var delPatient = delPatient || {};
+
+delPatient.deletePatientCreationDialog = null;
+
+/**
+ * Functions used to Delete a Patient (uses DeletePatientFragmentController)
+ */
+
+delPatient.returnUrl = "/coreapps/findpatient/findPatient.page?app=coreapps.findPatient";
+
+delPatient.createDeletePatientCreationDialog = function() {
+    delPatient.deletePatientCreationDialog = emr.setupConfirmationDialog({
+        selector: '#delete-patient-creation-dialog',
+        actions: {
+            confirm: function() {
+                var deleteReason = jQuery('#delete-reason').val().trim(); //Retrieve text from text box
+                if(deleteReason && deleteReason.length > 0) { //Should not be invalid or empty
+                    jq.ajax({
+                        url: '/openmrs/ws/rest/v1/patient/'+delPatient.patientUUID,
+                        type: 'DELETE',
+                        body: {
+                            "void_reason": deleteReason
+                        },
+                        success: function() {
+                            emr.successMessage("Success");
+                            jq('#delete-patient-creation-dialog' + ' .icon-spin').css('display', 'inline-block').parent().addClass('disabled');
+                            delPatient.deletePatientCreationDialog.close();
+                            jq(setTimeout(delPatient.goToReturnUrl, 1300)); //Allow the success message to display before redirecting to another page
```

Comment:
I haven't looked at it.

On Thu, May 5, 2016 at 9:37 AM, Shreyans Sheth <notifications@github.com>
wrote:

> In omod/src/main/webapp/resources/scripts/custom/deletePatient.js
> <https://github.com/openmrs/openmrs-module-coreapps/pull/20#discussion_r62214313>
> :
>
> > +        selector: '#delete-patient-creation-dialog',
> > +        actions: {
> > +            confirm: function() {
> > +                var deleteReason = jQuery('#delete-reason').val().trim(); //Retrieve text from text box
> > +                if(deleteReason && deleteReason.length > 0) { //Should not be invalid or empty
> > +                    jq.ajax({
> > +                        url: '/openmrs/ws/rest/v1/patient/'+delPatient.patientUUID,
> > +                        type: 'DELETE',
> > +                        body: {
> > +                            "void_reason": deleteReason
> > +                        },
> > +                        success: function() {
> > +                            emr.successMessage("Success");
> > +                            jq('#delete-patient-creation-dialog' + ' .icon-spin').css('display', 'inline-block').parent().addClass('disabled');
> > +                            delPatient.deletePatientCreationDialog.close();
> > +                            jq(setTimeout(delPatient.goToReturnUrl, 1300)); //Allow the success message to display before redirecting to another page
>
> Ahh thanks @djazayeri <https://github.com/djazayeri> [image: :smile:]
> Does everything else look Ok?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/openmrs/openmrs-module-coreapps/pull/20/files/dc9fe6feff9124c03eb69c49e7f755deaa7bd436#r62214313>
>